packages/backend/src/managers/playground/McpServerManager.ts

```diff
@@ -15,52 +15,67 @@
-// Cases 1 and 2 are covered by JsonWatcher, but case 3 is not.
-// For 2 and 3, we need to implement a UI to manage the servers.
-export class McpServerManager implements Disposable {
+export class McpServerManager extends Publisher<McpSettings> implements Disposable {
   private readonly settingsFile: string;
+  private mcpSettings: McpSettings;
+  private readonly _onUpdate: EventEmitter<McpSettings>;
+  readonly #jsonWatcher: JsonWatcher<McpSettings>;
 
-  constructor(private appUserDirectory: string) {
+  constructor(
+    rpcExtension: RpcExtension,
+    private appUserDirectory: string,
+  ) {
+    super(rpcExtension, MSG_MCP_SERVERS_UPDATE, () => this.getMcpSettings());
     this.settingsFile = path.join(this.appUserDirectory, MCP_SETTINGS);
+    this.mcpSettings = {
+      servers: {},
+    };
+    this._onUpdate = new EventEmitter<McpSettings>();
+    this.#jsonWatcher = new JsonWatcher<McpSettings>(this.settingsFile, { ...this.mcpSettings });
+    this.#jsonWatcher.onContentUpdated(this.onMcpSettingsUpdated.bind(this));
+    this.#jsonWatcher.init();
   }
 
-  async load(): Promise<McpServer[]> {
-    const mcpServers: McpServer[] = [];
-    try {
-      const mcpSettingsContent = await promises.readFile(this.settingsFile, 'utf8');
-      const mcpSettings = JSON.parse(mcpSettingsContent) as McpSettings;
-      if (!mcpSettings?.servers) {
-        return mcpServers;
+  private onMcpSettingsUpdated(mcpSettings: McpSettings): void {
+    this.mcpSettings = { servers: {} };
+    for (const [name, mcpServer] of Object.entries(mcpSettings.servers ?? {})) {
+      mcpServer.name = name;
+      if (!Object.values(McpServerType).includes(mcpServer.type)) {
+        console.warn(`McpServerManager: Invalid MCP server type ${mcpServer.type} for server ${mcpServer.name}.`);
+        continue;
       }
-      for (const entry of Object.entries(mcpSettings.servers)) {
-        const mcpServer: McpServer = entry[1] as McpServer;
-        mcpServer.name = entry[0];
-        if (!Object.values(McpServerType).includes(mcpServer.type)) {
-          console.warn(`McpServerManager: Invalid MCP server type ${mcpServer.type} for server ${mcpServer.name}.`);
-          continue;
-        }
-        mcpServers.push(mcpServer);
-      }
-    } catch (error: unknown) {
-      console.error(`McpServerManager: Failed to load MCP settings: ${error}`);
+      this.mcpSettings.servers[name] = mcpServer;
     }
-    return mcpServers;
+    this.notify();
+  }
+
+  getMcpSettings(): McpSettings {
+    return this.mcpSettings;
+  }
+
+  async toMcpClients(): Promise<McpClient[]> {
+    const enabledServers = Object.values(this.mcpSettings.servers).filter(server => server.enabled);
+    return toMcpClients(...enabledServers);
   }
 
   dispose(): void {
     // NO OP
   }
+
+  override notify(): void {
+    super.notify();
+    this._onUpdate.fire(this.getMcpSettings());
+  }
```

Comment:
If another class need to subscribe to the MCPManager, to subscribe to it, maybe we should port https://github.com/podman-desktop/extension-podman-quadlet/pull/539 to AI Lab.